### PR TITLE
completed assessment

### PR DESCRIPTION
**Pull request by sayoung0820**
# Submitting Week 1 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [✅] Coding Challenges
- [✅ ] Coding Concepts
- [✅ ] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

Question #3 on code-challenges: How could I have gotten the code to return the data without the commas? Thank you!

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -1,68 +1,168 @@
...
+// 5. Use a conditional to make an evaluation on the temperature of the water
+
+// if the temperature of the water evaluates to below 212 degrees, it's below boiling point
+// if the temperature of the water evaluates to equal to 212 degrees, it's at boiling point
+// if the temperature of the water evaluates to above 212 degrees, it's above boiling point
+// we need an else if none of the above is true, it will return an error
+
+// const water = (waterTemp) => {
+//     if(waterTemp === 212) {
+//         return "212 is at boiling point"
```
Comment (elysemontano) [inline code-challenges.js:70]: Consider using string interpolation here to make the response more dynamic

File: code-challenges.js [hunk 2]
```diff
@@ -1,68 +1,168 @@
...
+//         return "42 is below boiling point"
+//     } else {
+//         return "Sorry that is not a valid temp"
+//     }
+// }
+
+// console.log(water(212))
+// console.log(water(350))
+// console.log(water(42))
+// console.log(water("test"))
```
Comment (elysemontano) [inline code-challenges.js:83]: Great work adding a possible edge case!

File: code-challenges.js [hunk 3]
```diff
@@ -1,68 +1,168 @@
...
+// 1. Create a variable called padresRunsarray that will combine the values of two arrays then return the length
+// 2. The function uses the .length property
+// 3. The .length property looks at the index then returns the number of values in the array
+
+
+// const padres1984WorldSeriesRuns = [2, 5, 2, 2, 4]
+// const padres1998WorldSeriesRuns = [6, 3, 5, 3]
+// // Expected output: 9
+
+// var padresRunsArray = [2, 5, 2, 2, 4, 6, 3, 5, 3,]
```
Comment (elysemontano) [inline code-challenges.js:101]: Try using built in methods to combine the padres1984WorldSeriesRuns and padres1998WorldSeriesRun variables together.

File: code-challenges.js [hunk 4]
```diff
@@ -1,68 +1,168 @@
...
+// 2. Convert the string into an array using the split method
+// 3. Use the reverse method to reverse the values
+
+// const currentCohort = "Golf 2022"
 // Expected output: "2202 floG"
 
-// --------------------4) Create the code that will return the last index of the given value from the array using the test variables provided below.
+// var curCohort = ("Golf 2022")
+
+// var curCohort = ("Golf 2022".split(""))
```
Comment (elysemontano) [inline code-challenges.js:120]: You asked about this problem and how to get the return without the commas, did you mean quotes?  If so, split needs the quotes in here so that it splits each character is it's own string.  Without quotes the array would just wrap around the string ["Golf 2022"].  I find it's helpful to just console.log different variations of arguments with split.  Ex: console.log("hello world".split(" ")).

File: code-challenges.js [hunk 5]
```diff
@@ -1,68 +1,168 @@
...
+
+
+
+
+// // --------------------5) Create the code that will sort all the numbers in an array from largest to smallest. Use the test variables provided below.
+
+// // Describe your process:
+// 1. Create a function called numbers
+// 2. Set up two paramaters called numbers1 and numbers2
+// 3. Use the .sort built in method
```
Comment (elysemontano) [inline code-challenges.js:154]: Make sure to expand on this if you are providing extra information to the sort method that handles some kind of logic.

File: coding-concepts.js [hunk 6]
```diff
@@ -6,49 +6,63 @@
...
-// console.log(weekendDays.toUpperCase())
+// const weekendDays = ["saturday", "sunday"]
+// // console.log(weekendDays.toUpperCase())
+
+// console.log("saturday, sunday".toUpperCase())
 
 // a) Your answer:
+    // This should return "SATURDAY, SUNDAY"
 // b) Verify and explain:
+    // Answer: Did not log what I thought it would because it did not have the correct function in the console log
```
Comment (elysemontano) [inline coding-concepts.js:57]: This is often a confusing error, but note what kind of error it says "Type error".  When we run into this error, it is actually telling us we are trying to use a built in method that does not work on this data type.  In this case, toUpperCase() is a string method, not an array method.

File: interview-questions.md [hunk 7]
```diff
@@ -8,50 +8,52 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-Researched answer:
+Researched answer: A git is a system that allows collobarators to simultaneously work on projects at the same time. Each coder can add changes then push them up to Github. Github is a cloud based service that hosts the main projects in repositories, allowing the project to be better managed by all colaborators.
 
 2. Which JavaScript operators will return a Boolean value?
 
-Your answer:
+Your answer: The loose equality operator and the strict equality operator
 
-Researched answer:
+Researched answer: Boolean values are returned when using the strict or loose equality operators. The loose equality operator converts the data type to make the equal first, then makes a comparison. The strict equality operator requires that the types of data be the same before returning a boolean.
```
Comment (elysemontano) [inline interview-questions.md:27]: Don't forget about relational and logical operators

File: interview-questions.md [hunk 8]
```diff
@@ -8,50 +8,52 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 3. What is an index? What is the difference between index and value?
 
-Your answer:
+Your answer: An index is a placeholder for a character in a string
```
Comment (elysemontano) [inline interview-questions.md:31]: Make sure to expand on your answers.  This is supposed to reflect how you would respond to an interviewer in an interview.  Short answers do not give you the chance to share your knowledge.  Try providing examples and digging deeper into the definitions.

**Review by elysemontano — APPROVED** [review]
Nice work Shannon!  I left some feedback throughout the assessment to take a look at.
Current assessment level is novice - You did a really great job on the coding-concepts section.  Make sure to expand on your interview answers.  For the coding challenges, really focus on reviewing built-in methods and functions in the syllabus and how to make code more dynamic.
